tasks/ts.ts

```diff
@@ -0,0 +1,76 @@
+import * as path from 'path';
+
+export = function(grunt: IGrunt) {
+	const distDirectory = grunt.config.get<string>('distDirectory');
+	const defaultOptions: any = {
+		dist: {
+			exclude: ['tests/**/*.ts'],
+			compilerOptions: {
+				outDir: distDirectory,
+				mapRoot: path.join('..', distDirectory, '_debug'),
+				declaration: true
+			}
+		},
+		esm: {
+			exclude: ['tests/**/*.ts'],
+			compilerOptions: {
+				target: 'es6',
+				module: 'es6',
+				sourceMap: false,
+				outDir: 'dist/esm',
+				inlineSourceMap: true,
+				inlineSources: true
+			}
+		}
+	};
+	const writeOptions = {
+		encoding: 'UTF8'
+	};
+
+	grunt.registerTask('ts', <any> function () {
+		grunt.loadNpmTasks('grunt-ts');
+		const flags = this.flags && Object.keys(this.flags).length ? this.flags : { 'dev': true };
+		const tsconfig = grunt.config.get<any>('tsconfig');
+		const tsOptions = grunt.config.get<any>('ts');
+		const baseOptions = {
+			failOnTypeErrors: true,
+			fast: 'never'
+		};
+
+		grunt.config.set('ts.options', baseOptions);
+
+		Object.keys(flags).forEach((key) => {
+			if (flags[key]) {
```

Comment:
using `this.args` which returns an `array` means we won't need the Object.keys or this `if` condition which should neaten things up and remove a level of indentation.